spaceship/lib/spaceship/connect_api/models/app.rb

```diff
@@ -223,15 +223,22 @@ def ensure_version!(version_string, platform: nil, client: nil)
         end
       end
 
-      def get_latest_app_store_version(client: nil, platform: nil, includes: nil)
+      # This will return the latest version that is available on the App Store.
+      # By default, when limit is nil, it will eagerly fetch all versions, sort them, and then return the last one.
+      # If you set a limit, it will only fetch the latest versions (which is much faster), and then return the last one.
+      # However, if Apple changes the way they sort versions, this might not return the latest versions, hence why
+      # the default behavior is to fetch all versions.
+      # Unfortunately, there's no way to sort by version number on the server side, so we have to do it on the client side.
+      # Pass in a limit knowing the risks of this changing silently in the future.
```

Comment:
IMO the risk that Apple will start returning App Store versions in a different order with `limit` provided is so infinitesimally small that I propose a default value of `limit = 1`

Reply:
If you want as a safeguard we could add an environment variable to override this behavior if it causes problems

Reply:
I don't mind the default value being `limit=1`  as it doesn't change the behavior for today's users and bring the performance improvement to all users.